neofetch

```diff
@@ -2189,29 +2189,36 @@ get_wm_theme() {
 get_cpu() {
     case $os in
         "Linux" | "MINIX" | "Windows")
-            # Get CPU name.
+            dt="/proc/device-tree"
             cpu_file="/proc/cpuinfo"
 
-            case $kernel_machine in
-                "frv" | "hppa" | "m68k" | "openrisc" | "or"* | "powerpc" | "ppc"* | "sparc"*)
-                    cpu="$(awk -F':' '/^cpu\t|^CPU/ {printf $2; exit}' "$cpu_file")"
-                ;;
-
-                "s390"*)
-                    cpu="$(awk -F'=' '/machine/ {print $4; exit}' "$cpu_file")"
-                ;;
-
-                "ia64" | "m32r")
-                    cpu="$(awk -F':' '/model/ {print $2; exit}' "$cpu_file")"
-                    [[ -z "$cpu" ]] && cpu="$(awk -F':' '/family/ {printf $2; exit}' "$cpu_file")"
-                ;;
-
-                *)
-                    cpu="$(awk -F '\\s*: | @' \
-                            '/model name|Hardware|Processor|^cpu model|chip type|^cpu type/ {
-                            cpu=$2; if ($1 == "Hardware") exit } END { print cpu }' "$cpu_file")"
-                ;;
-            esac
+            if [[ ! -r $dt ]]
+            then
+               # Get CPU name.
+    
+               case $kernel_machine in
+                   "frv" | "hppa" | "m68k" | "openrisc" | "or"* | "powerpc" | "ppc"* | "sparc"*)
+                       cpu="$(awk -F':' '/^cpu\t|^CPU/ {printf $2; exit}' "$cpu_file")"
+                   ;;
+    
+                   "s390"*)
+                       cpu="$(awk -F'=' '/machine/ {print $4; exit}' "$cpu_file")"
+                   ;;
+    
+                   "ia64" | "m32r")
+                       cpu="$(awk -F':' '/model/ {print $2; exit}' "$cpu_file")"
+                       [[ -z "$cpu" ]] && cpu="$(awk -F':' '/family/ {printf $2; exit}' "$cpu_file")"
+                   ;;
+    
+                   *)
+                       cpu="$(awk -F '\\s*: | @' \
+                               '/model name|Hardware|Processor|^cpu model|chip type|^cpu type/ {
+                               cpu=$2; if ($1 == "Hardware") exit } END { print cpu }' "$cpu_file")"
+                   ;;
+               esac
+        else
+                cpu="$(cat $dt/model)"
```

Comment:
Weird... `/proc/device-tree` doesn't exist for 5/6 of my devices on very new kernels, and for one that does exist, the model file returns the name of the SoC `OrangePi 3 LTS` instead of the CPU name. What's the requirement for `/proc/device-tree` to exist?

<img width="258" alt="image" src="https://user-images.githubusercontent.com/22280294/200383629-5c227313-8f96-4a1e-98c6-8970ee59a59e.png">

<img width="467" alt="image" src="https://user-images.githubusercontent.com/22280294/200383339-4c359d05-f7b4-42a8-80bf-fda4fe3b2d88.png">

<img width="391" alt="image" src="https://user-images.githubusercontent.com/22280294/200383369-e50dbd72-2cba-4be2-ba0a-4337a71746a5.png">

<img width="402" alt="image" src="https://user-images.githubusercontent.com/22280294/200383969-e4df86df-7657-4e01-aa37-759b54fa7055.png">

<img width="467" alt="image" src="https://user-images.githubusercontent.com/22280294/200384318-264ba839-38a2-4e7c-8fef-b5038cc9d76c.png">

<img width="407" alt="image" src="https://user-images.githubusercontent.com/22280294/200383314-d9dd127d-9ad2-47c2-a6ee-c8d6399caf32.png">

Reply:
In addition, it's possible to parse `compatible` property and show processor codename in kernel, but it's not always true and in some cases will not work. For example, `compatible` property contains specific components for different SoCs or only one entry without the SoC codename and so.